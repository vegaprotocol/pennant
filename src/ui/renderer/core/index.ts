export * from "./abstract-renderer";
export * from "./batch/batch-geometry";
export * from "./geometry/geometry";
export * from "./geometry/geometry-system";
export * from "./renderer";
export * from "./textures/base-texture";
export * from "./textures/texture";
