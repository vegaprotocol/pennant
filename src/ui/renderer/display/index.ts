export * from "./container";
export * from "./display-object";
