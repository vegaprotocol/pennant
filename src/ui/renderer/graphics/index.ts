export * from "./graphics";
export * from "./graphics-data";
export * from "./graphics-geometry";
export * from "./graphics-renderer";
