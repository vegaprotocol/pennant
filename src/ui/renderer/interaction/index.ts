export * from "./interaction-data";
export * from "./interaction-event";
export * from "./interaction-tracking-data";
export * from "./interactive-target";
