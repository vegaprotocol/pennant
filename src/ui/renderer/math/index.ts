import { Area } from "./shapes/area";
import { Circle } from "./shapes/circle";
import { Line } from "./shapes/line";
import { Polygon } from "./shapes/polygon";
import { Rectangle } from "./shapes/rectangle";

export * from "./const";
export * from "./matrix";
export * from "./point";
export * from "./transform";

export { Area };
export { Circle };
export { Line };
export { Polygon };
export { Rectangle };

export type Shape = Area | Circle | Line | Polygon | Rectangle;
