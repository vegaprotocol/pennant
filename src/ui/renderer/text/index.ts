export * from "./text";
export * from "./text-metrics";
export * from "./text-style";
