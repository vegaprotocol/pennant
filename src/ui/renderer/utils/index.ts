export * from "./color/hex";
export * from "./data/removeItems";
export * from "./data/sign";
