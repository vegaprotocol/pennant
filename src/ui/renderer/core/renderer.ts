import { GraphicsRenderer } from "../graphics";
import { InteractionManager } from "../interaction/interaction-manager";
import { Matrix } from "../math";
import { SpriteRenderer } from "../sprite/sprite-renderer";
import { AbstractRenderer, RendererOptions } from "./abstract-renderer";
import { CanvasContextSystem } from "./canvas-context-system";
import { CanvasMaskSystem } from "./canvas-mask-system";
import { GeometrySystem } from "./geometry/geometry-system";
import { RenderableObject } from "./renderable-object";

const tempMatrix = new Matrix();

export class Renderer extends AbstractRenderer {
  public mask: CanvasMaskSystem | null = null;
  public canvasContext: CanvasContextSystem | null = null;
  public context: CanvasRenderingContext2D;
  public geometry: GeometrySystem | null;

  private _projTransform: Matrix | null;

  constructor(options?: RendererOptions) {
    super(options);

    this.context = this.view.getContext("2d") as CanvasRenderingContext2D;
    this.context.imageSmoothingEnabled = true;
    this.geometry = null;

    this.plugins.sprite = new SpriteRenderer(this);
    this.plugins.graphics = new GraphicsRenderer(this);
    this.plugins.interaction = new InteractionManager(this);

    this._projTransform = null;

    this.mask = new CanvasMaskSystem(this);
    this.canvasContext = new CanvasContextSystem(this);

    this.resize(options?.width ?? 800, options?.height ?? 600);
  }

  public render(displayObject: RenderableObject) {
    const context = this.context;

    this._lastObjectRendered = displayObject;

    displayObject.updateTransform();

    context.save();
    context.setTransform(1, 0, 0, 1, 0, 0);
    context.globalAlpha = 1;

    context.clearRect(0, 0, this.width, this.height);

    /*   context.fillStyle = this._backgroundColorString;
    context.fillRect(0, 0, this.width, this.height);
  */
    displayObject.render(this);

    context.restore();
  }

  public clear(
    clearColor: string = this._backgroundColorString,
    alpha: number = this.backgroundAlpha
  ): void {
    const context = this.context;

    context.clearRect(0, 0, this.width, this.height);

    /*     if (clearColor) {
      context.globalAlpha = alpha;
      context.fillStyle = clearColor;
      context.fillRect(0, 0, this.width, this.height);
      context.globalAlpha = 1;
    } */
  }

  setContextTransform(
    transform: Matrix,
    roundPixels?: boolean,
    localResolution?: number
  ): void {
    let mat = transform;
    const proj = this._projTransform;
    const resolution = 1;

    localResolution = localResolution || resolution;

    if (proj) {
      mat = tempMatrix;
      mat.copyFrom(transform);
      mat.prepend(proj);
    }

    if (roundPixels) {
      this.context.setTransform(
        mat.a * localResolution,
        mat.b * localResolution,
        mat.c * localResolution,
        mat.d * localResolution,
        (mat.tx * resolution) | 0,
        (mat.ty * resolution) | 0
      );
    } else {
      this.context.setTransform(
        mat.a * localResolution,
        mat.b * localResolution,
        mat.c * localResolution,
        mat.d * localResolution,
        mat.tx * resolution,
        mat.ty * resolution
      );
    }
  }
}
