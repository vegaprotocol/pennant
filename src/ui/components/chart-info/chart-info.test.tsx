import renderer from "react-test-renderer";

import { ChartInfo } from ".";

test("ChartInfo renders correctly", () => {
  const component = renderer.create(
    <ChartInfo bounds={[new Date(2021, 2, 1), new Date(2021, 2, 2)]} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
