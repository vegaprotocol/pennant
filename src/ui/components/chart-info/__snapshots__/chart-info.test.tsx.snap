// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartInfo renders correctly 1`] = `
<div
  className="chart-info-wrapper"
>
  <div>
    <span>
      00:00 01 Mar 2021
    </span>
    <span
      className="text-muted"
    >
       to 
    </span>
    <span>
      00:00 02 Mar 2021
    </span>
  </div>
</div>
`;
